checkov/terraform/checks/resource/gcp/GKEUseCosImage.py

```diff
@@ -12,24 +12,36 @@ def __init__(self):
 
     def scan_resource_conf(self, conf):
         """
-            Looks for password configuration at azure_instance:
-            https://www.terraform.io/docs/providers/google/r/compute_ssl_policy.html
-        :param conf: google_compute_ssl_policy configuration
-        :return: <CheckResult>
+        CKV_GCP_22
+        error creating NodePool: googleapi: Error 400: Creation of node pools using node images based on Docker
+        container runtimes is not supported in GKE v1.24+ clusters as Dockershim has been removed in Kubernetes v1.24.
+
+        image_type   = "COS"
         """
-        if 'node_config' in conf:
-            node_config = conf.get('node_config', [{}])[0]
-            self.evaluated_keys = ['node_config']
-            if not isinstance(node_config, dict):
-                return CheckResult.UNKNOWN
-
-            if conf.get('node_config', [{}])[0].get('image_type', [''])[0].lower().startswith('cos'):
-                self.evaluated_keys = ['node_config/[0]/image_type']
-                return CheckResult.PASSED
-            if conf.get('remove_default_node_pool', [{}])[0]:
-                self.evaluated_keys.append('remove_default_node_pool')
-                return CheckResult.PASSED
-        return CheckResult.FAILED
+        if conf.get('version') and isinstance(conf.get('version'), list):
+            raw = conf.get('version')[0]
+            splitter = raw.split(".")
+
+            if len(splitter) >= 2:
+                version = float(splitter[0] + "." + splitter[1])
+                if version >= 1.24:
```

Comment:
a better way to compare versions is using an existing python package to do it,
for example I saw [this thread ](https://stackoverflow.com/questions/11887762/how-do-i-compare-version-numbers-in-python)